src/main/java/com/Repository/AssociationRepository.java

```diff
@@ -2,15 +2,22 @@
 
 import com.Model.Event;
 import com.Model.PatientEventAssociation;
-import org.springframework.data.domain.Page;
-import org.springframework.data.domain.PageRequest;
+import com.Model.Place;
 import org.springframework.data.jpa.repository.Query;
-import org.springframework.data.repository.CrudRepository;
+import org.springframework.data.repository.PagingAndSortingRepository;
+import org.springframework.data.rest.core.annotation.RepositoryRestResource;
 
-public interface AssociationRepository extends CrudRepository<PatientEventAssociation, Long> {
+@RepositoryRestResource(collectionResourceRel = "PatientEventAssociation", path = "association")
+public interface AssociationRepository extends PagingAndSortingRepository<PatientEventAssociation, Long> {
 
     @Query("SELECT DISTINCT p.event FROM PatientEventAssociation p " +
             "WHERE p.isValid=true")
-    Page<Event> findAllValidEvents (PageRequest pageNum);
+    Iterable<Event> findAllValidEvents ();
 
+    @Query("SELECT p.event.place FROM PatientEventAssociation p " +
```

Comment:
why do you need to specify these queries? The correct naming of the method should do the job, and it is safer.

Reply:
I tried to find correct naming, but seems like Crudrepository does not provide naming methods for retrieving nested objects. I will do more research.